src/autocat/saa.py

```diff
@@ -310,33 +325,34 @@ def substitute_single_atom_on_surface(
         substitution functionality is folded into a more general form.
 
     """
-    tags = host_structure.get_tags()
-    constraints = host_structure.constraints
```

Comment:
We needed to store the `tags` and `constraints` separately since they were previously lost when going from `ase.Atoms` -> `pymatgen.Structure` -> `ase.Atoms` to use the `pymatgen` substitution function. 

But since we are now manipulating the `ase.Atoms` object directly without converting to `pymatgen.Structure` (except just to do structure matching check) we no longer need to store them as separate variables

Reply:
Ah, got it. Thanks, that makes sense.